exchange/exchange.go

```diff
@@ -447,7 +450,17 @@ func (e *exchange) getAllBids(
 			var reqInfo adapters.ExtraRequestInfo
 			reqInfo.PbsEntryPoint = bidderRequest.BidderLabels.RType
 			reqInfo.GlobalPrivacyControlHeader = globalPrivacyControlHeader
-			bids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest.BidRequest, bidderRequest.BidderName, adjustmentFactor, conversions, &reqInfo, accountDebugAllowed, headerDebugAllowed)
+
+			//map imp to stored request data if exists
+			hasStoredReq, impsToStoredReq, parseErr := parseStoredImp(bidderRequests)
```

Comment:
The tests are saying this line introduces a data race.

Reply:
Modified.